### ISSUE TITLE
long long missing in adios_types.h

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS#187]
Although `long long` and `unsigned long long` are [basic/fundamental C (& C++) types](https://en.wikipedia.org/wiki/C_data_types#Basic_types), there are only:
```C
                     ,adios_byte = 0                 /* (1) */
                     ,adios_short = 1                /* (2) */
                     ,adios_integer = 2              /* (4) */
                     ,adios_long = 4                 /* (8) */

                     ,adios_unsigned_byte = 50       /* (1) */
                     ,adios_unsigned_short = 51      /* (2) */
                     ,adios_unsigned_integer = 52    /* (4) */
                     ,adios_unsigned_long = 54 /* (8) */
```

in [adios_types.h](https://github.com/ornladios/ADIOS/blob/v1.13.1/src/public/adios_types.h)

Can you add support for `adios_long_long` and `adios_unsigned_long_long`?

Also, the `(size)` comment in those lines is platform specific.

cc @pnorbert

**Comment by ax3l** [comment]
Ok, I think I understand what you did there.

`adios_short` is actually a *platform independent* `adios_int16_t`, `adios_integer` a *platform independent* `adios_int32_t` and `adios_long` a *platform independent* `adios_int64_t`?

(Same for unsigned and floating points.)

**Comment by pnorbert** [comment]
Exactly. This approach, however, put the task on the user to make sure the
correct type is chosen. In adios 2.x, the C++ API lets the user to use the
C++ types.

**Comment by ax3l** [comment]
@pnorbert alright, thanks! And you assume that a Byte is 8 Bit, otherwise it's not portable? Just checking the corner cases :)

**Comment by pnorbert** [comment]
in adios 1.x, adios_byte simply means 8 bit. The user needs to make the
mapping between memory type to the adios type. You would need to store a 16
bit wide character string as a adios_short array manually.

**Comment by ax3l** [comment]
@pnorbert alright, thanks.

Another question: what's the recommended type in adios 1 to represent C++ `bool`s? We are currently mapping them to `unsigned char`s, but allowing a `bool` equivalent in meta data would be great to differentiate properly in reads.

**Comment by pnorbert** [comment]
Well, just that, unsigned char.
